tests/pytest_loganalyzer/pytest_loganalyzer.py

```diff
@@ -0,0 +1,181 @@
+import sys
+import logging
+import os
+import re
+
+from time import gmtime, strftime
+from os.path import join, split
+from os.path import normpath
+
+# Integration with legacy loganalyzer
+ANSIBLE_LOGANALYZER = normpath(join(split(__file__)[0], "../../ansible/roles/test/files/tools/loganalyzer/loganalyzer.py"))
+sys.path.append(split(ANSIBLE_LOGANALYZER)[0])
+from loganalyzer import LogAnalyzer as ansible_loganalizer
+
+COMMON_MATCH = "{}".format(join(split(ANSIBLE_LOGANALYZER)[0], "loganalyzer_common_match.txt"))
+COMMON_IGNORE = "{}".format(join(split(ANSIBLE_LOGANALYZER)[0], "loganalyzer_common_ignore.txt"))
+COMMON_EXPECT = "{}".format(join(split(ANSIBLE_LOGANALYZER)[0], "loganalyzer_common_expect.txt"))
+SYSLOG_TEMP_FOLDER = "/tmp/pytest-run/syslog"
+
+
+class PytestLogAnalyzer:
+    def __init__(self, ansible_host, marker_prefix, run_dir="/tmp"):
+        self.ansible_host = ansible_host
+        self.run_dir = run_dir
+        self.run_id = None
+        self.extracted_syslog = os.path.join(self.run_dir, "syslog")
+        self.marker_prefix = marker_prefix
+
+        self.match_regex = []
+        self.expect_regex = []
+        self.ignore_regex = []
+        self._analyzed = True
+
+    def update_marker_prefix(self, marker_prefix):
+        """
+        @summary: Update configured marker prefix
+        """
+        self.marker_prefix = marker_prefix
+
+    def load_common_config(self):
+        """
+        @summary: Load regular expressions from common files, which are localted in folder with legacy loganalyzer.
+                  Loaded regular expressions are used by "analyze" method to match expected text in the downloaded log file.
+        """
+        analyzer = ansible_loganalizer(self.run_id, False)
+        self.match_regex = analyzer.create_msg_regex([COMMON_MATCH])[1]
+        self.ignore_regex = analyzer.create_msg_regex([COMMON_IGNORE])[1]
+        self.expect_regex = analyzer.create_msg_regex([COMMON_EXPECT])[1]
+
+    def parse_regexp_file(self, src):
+        """
+        @summary: Get regular expressions defined in src file.
+        """
+        analyzer = ansible_loganalizer(self.run_id, False)
+        return analyzer.create_msg_regex([src])[1]
+
+    def run_cmd(self, callable, *args, **kwargs):
+        """
+        @summary: Initialize loganalyzer, execute function and analyze syslog.
+
+        @param callable: Python callable or function to be executed.
+        @param args: Input arguments for callable function.
+        @param kwargs: Input arguments for callable function.
+
+        @return: Tuple of two items: (callable(*args, **kwargs) -> VALUE, self.analyze() -> dict)
+        """
+        self.init()
+        call_result = callable(*args, **kwargs)

```

Comment:
If callable throws, do we need to return log analysis report?

Reply:
Updated. If callable throw exception - return empty result (None, {}).

Reply:
How can one distinguish between two cases:
1. callable returns None
2. callable throws and run_cmd returns None

suggest to return 3-tuple (return, exception, report)

Reply:
BTW, callable is a good name in general but a bad name for python variable (callable is builtin python function). Could you please rename?

Reply:
Fixed.